src/renderers/shared/ReactDebugTool.js

```diff
@@ -279,26 +299,78 @@ var ReactDebugTool = {
     checkDebugID(debugID);
     checkDebugID(parentDebugID, true);
     emitEvent('onBeforeMountComponent', debugID, element, parentDebugID);
+    if (isProfiling && hasAdvancedConsoleFeatures) {
+      var isCompositeComponent = typeof element.type === 'function';
+      if (isCompositeComponent) {
+        var displayName = ReactComponentTreeHook.getDisplayName(debugID);
+        var label = displayName + ' (mount)';
+        console.time(label);
+      }
+    }
   },
   onMountComponent(debugID) {
     checkDebugID(debugID);
     emitEvent('onMountComponent', debugID);
+    if (isProfiling && hasAdvancedConsoleFeatures) {
+      var element = ReactComponentTreeHook.getElement(debugID);
+      var isCompositeComponent = typeof element.type === 'function';
+      if (isCompositeComponent) {
+        var displayName = ReactComponentTreeHook.getDisplayName(debugID);
+        var label = displayName + ' (mount)';
+        console.timeEnd(label);
+      }
+    }
   },
   onBeforeUpdateComponent(debugID, element) {
     checkDebugID(debugID);
     emitEvent('onBeforeUpdateComponent', debugID, element);
+    if (isProfiling && hasAdvancedConsoleFeatures) {
+      var isCompositeComponent = typeof element.type === 'function';
+      if (isCompositeComponent) {
+        var displayName = ReactComponentTreeHook.getDisplayName(debugID);
+        var label = displayName + ' (update)';
+        console.time(label);
+      }
+    }
   },
   onUpdateComponent(debugID) {
     checkDebugID(debugID);
     emitEvent('onUpdateComponent', debugID);
+    if (isProfiling && hasAdvancedConsoleFeatures) {
+      var element = ReactComponentTreeHook.getElement(debugID);
+      var isCompositeComponent = typeof element.type === 'function';
```

Comment:
`ReactTestUtils` checks an instance, not an element, so that’s why it’s different.
I could add a convenience method to the hook but I’d wait for a little more duplication first.